osb-backup-manager/src/main/java/de/evoila/cf/backup/service/manager/BackupServiceManager.java

```diff
@@ -128,9 +137,39 @@ private void executeBackup(BackupExecutorService backupExecutorService, Endpoint
                 i++;
             }
 
+            for(CompletableFuture completionFuture : completionFutures) {
```

Comment:
@patsys changed the check. the delete function is now called when all threads are done. working on a timeout right now

Reply:
why retention not in line 128.

Reply:
Every item is handled by its own thread, line 128 only checks the success of ONE single thread